src/main/scala/org/broadinstitute/dsde/workbench/sam/api/SamModelDirectives.scala

```diff
@@ -7,15 +7,16 @@ import akka.http.scaladsl.unmarshalling.FromRequestUnmarshaller
 import org.broadinstitute.dsde.workbench.model._
 import org.broadinstitute.dsde.workbench.sam._
 import org.broadinstitute.dsde.workbench.sam.service.UserService
+import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
 
 /**
   * Created by gpolumbo on 3/26/2018
   */
 trait SamModelDirectives {
   val userService: UserService
 
-  def withSubject(email: WorkbenchEmail): Directive1[WorkbenchSubject] =
-    onSuccess(userService.getSubjectFromEmail(email)).map {
+  def withSubject(email: WorkbenchEmail, samRequestContext: SamRequestContext = SamRequestContext(null)): Directive1[WorkbenchSubject] =
+    onSuccess(userService.getSubjectFromEmail(email, samRequestContext)).map {
```

Comment:
Do we want this to be a part of the trace? Trying to add tracing by changing the order leads to some errors (which may be easily solved if this is the behavior we want)

when reordering the code:

from
```
      withSubject(WorkbenchEmail(email)) { subject =>
        completeWithTrace(samRequestContext =>
          managedGroupService.addSubjectToPolicy(managedGroup.resourceId, accessPolicyName, subject, samRequestContext).map(_ => StatusCodes.NoContent))
      }
```
to
```
        completeWithTrace{samRequestContext =>
          withSubject(WorkbenchEmail(email)) { subject =>
            managedGroupService.addSubjectToPolicy(managedGroup.resourceId, accessPolicyName, subject, samRequestContext).map(_ => StatusCodes.NoContent)
          }
        }
```
results in this error:
```
[error] /Users/ajang/Firecloud/sam/src/main/scala/org/broadinstitute/dsde/workbench/sam/api/ManagedGroupRoutes.scala:123:126: type mismatch;
[error]  found   : scala.concurrent.Future[akka.http.scaladsl.model.StatusCodes.Success]
[error]  required: akka.http.scaladsl.server.RequestContext => scala.concurrent.Future[akka.http.scaladsl.server.RouteResult]
[error]             managedGroupService.addSubjectToPolicy(managedGroup.resourceId, accessPolicyName, subject, samRequestContext).map(_ => StatusCodes.NoContent)
[error]                                                                                                                              ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
```

Reply:
the problem with this and the `requireAction` stuff is that `withSubject` and `requireAction` are both `Directives` and `completeWithTrace` can't contain a `Directive`. 

I better approach would be instead of `completeWithTrace` would be to have a `withTrace` directive.

Reply:
```
def withSamRequestContext: Directive1[SamRequestContext] = traceRequest.map { span => SamRequestContext(span) }
```

Reply:
//implemented